libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,657 @@
+                item.listType = this.listType;
+                item.hasByLine = this.hasByLine;
+                item.noSeperator = this.noSeperator;
+                this.stateChanges.next(item);
+            });
+
+        });
+    }
+
+    ngOnDestroy(): void {
+        super.ngOnDestroy();
+
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+        }
+        if (this._dsSubscription) {
+            this._dsSubscription.unsubscribe();
+        }
+        if (this.selectionModel) {
+            this.selectionModel.changed.unsubscribe();
+        }
+        if (this._itemsSubscription) {
+            this._itemsSubscription.unsubscribe();
+        }
+    }
+
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+
+    set value(value: any) {
+        super.setValue(value);
+    }
+
+
+    /** @hidden */
+    /**handline keyboard operations
+    *  on list and list items
+    */
+    handleKeyDown(event: KeyboardEvent): boolean {
+        event.stopImmediatePropagation();
+        if (this.keyManager) {
+            if (event.keyCode === DOWN_ARROW || event.keyCode === UP_ARROW) {
+                return false;
+            } else if (event.keyCode === ENTER || event.keyCode === SPACE) {
+                this.updateNavigation(event);
+                return false;
+            }
+        }
+    }
+
+    _initializeDS(ds: FdpListDataSource<any>): void {
+        this.dsItems = [];
+        if (isDataSource(this.dataSource)) {
+            this.dataSource.close();
+            if (this._dsSubscription) {
+                this._dsSubscription.unsubscribe();
+                this._dsSubscription = null;
+            }
+        }
+        // Convert ListDataSource<T> | T[] as DataSource
+        this._dataSource = this._openDataStream(ds);
+    }
+
+    _toDataStream(ds: FdpListDataSource<any>): ListDataSource<any> {
+        if (isDataSource(ds)) {
+            return ds;
+        }
+        return undefined;
+    }
+
+    _openDataStream(ds: FdpListDataSource<any>): ListDataSource<any> {
+        const initDataSource = this._toDataStream(ds);
+        if (initDataSource === undefined) {
+            throw new Error(`[dataSource] source did not match an array, Observable, or DataSource`);
+        }
+        /**
+         * This is single point of data entry to the component. We dont want to set data on different
+         * places. If any new data comes in either you do a search and you want to pass initial data
+         * its here.
+         */
+        this._dsSubscription = initDataSource
+            .open()
+            .pipe(takeUntil(this._destroyed))
+            .subscribe((data) => {
+                this.dsItems = data || [];
+                this.stateChanges.next(this.dsItems);
+                this._cd.markForCheck();
+            });
+        // initial data fetch
+        initDataSource.match('*');
+        return initDataSource;
+    }
+
+    scrollHandler(): void {
```

Comment:
Can you cleanup a bit this code, for things that you want public keep it public (without `public` modifer), for some you have public for some you done. Also we sort them as it is specified by coding guidelines



Also for Private API (properties, methods) that needs to be used from template so they need to be public

use `_` prefix with jsdoc hidden. for things that should be private keep it private

Reply:
It is bound in template thats why I made it public, marking it to private is leading to compilation errors.